### DISCUSSION TITLE
setFieldValue 和 setFiledsValue 设置值的时候，导致的 validator 不一致

### DISCUSSION THREAD

**Discussion by LuckyFBB** [ant-design/ant-design#43882] [Q&A]
[v4 版本 demo](https://stackblitz.com/edit/react-c5gnh6?file=demo.tsx)

[v5 版本 demo](https://stackblitz.com/edit/react-msikpw?file=demo.tsx)

复现步骤，输入 Username 值，长度大于 5 时，会展示如下报错

<img width="459" alt="image" src="https://github.com/ant-design/ant-design/assets/38368040/b247dd54-c02a-4ceb-ae49-ed0d56f7bf25">

点击 useSetFieldValue select，会使用 setFieldValue 去重写 username 域的值，input 框的值更新了，但是报错依旧存在

<img width="453" alt="image" src="https://github.com/ant-design/ant-design/assets/38368040/4a67ffc8-ac64-4e65-a68e-467987fd0276">

点击 useSetFieldsValue select，会使用 setFieldsValue 去重写 username 域的值，input 框的值更新了，报错消失了

<img width="452" alt="image" src="https://github.com/ant-design/ant-design/assets/38368040/aabd8dca-6326-4bd7-8886-2a2d0b0bad5d">

<img width="460" alt="image" src="https://github.com/ant-design/ant-design/assets/38368040/e271ad20-0b89-4864-87e5-62b97902c8ad">

想问一下，有没有 bro 知道这是为什么？是我使用方式不对？还是是专门这样设计的？还是为一个 bug？

**Comment by zombieJ** [comment]
setFieldValue 内部也是 setFieldsValue 实现，估计是什么其他原因

**Reply by LuckyFBB** [reply]
那为什么会出现这样的情况呢？是我的使用方式不对么

**Comment by mortalYoung** [comment]
@zombieJ 不是其他原因导致的，就是 setFieldValue 和 setFieldsValue 的实现不一致导致的，具体你看看这里的代码:

https://github.com/react-component/field-form/blob/e9184848e73a5e39fd2976826a7020328dba01ed/src/useForm.ts#L761-L785
